hw-sockets/client_helper.py

```diff
@@ -0,0 +1,113 @@
+"""Helper classes for client part.
+
+MetricCollector: Class to collect metrics. You have to define a
+  function able to return value to use this class.
+
+Size: Class to handle file size and pretty print it.
+
+"""
+
+import functools
+from collections import OrderedDict
+from typing import Callable
+
+
+class MetricCollectorIsRunningError(Exception):
+    def __str__(self):
+        return "MetricCollector is already running"
+
+
+class MetricCollectorIsStoppedError(Exception):
+    def __str__(self):
+        return "MetricCollector is already stopped"
+
+
+class MetricCollector:
+    """Class to collect metrics
+
+    Args:
+      name: name of metric
+
+      target_function: function to call to get metric
+
+    """
+
+    def __init__(self, name: str, target_function: Callable):
+        self.name = name
+        self.target_function = target_function
+        self.value = None
+        self._active = False
+
+    def start_collect(self):
+        """Begin observe metric"""
+        if self._active:
+            raise MetricCollectorIsRunningError
+        else:
+            self._active = True
+
+        while self._active:
+            self.value = self.target_function()
```

Comment:
I suppose it would be better if we run `sleep` here for one second, for saving CPU usage